lib/note-detail.jsx

```diff
@@ -19,36 +19,61 @@ export default React.createClass( {
 
 	componentWillMount: function() {
 		this.queueNoteSave = debounce( this.saveNote, saveDelay );
-		this.noteEditor = uninitializedNoteEditor;
+		this.focus = () => this.refs.editor.focus();
+		this.state = {
+			editorState: EditorState.createEmpty( urlDecorator )
+		};
 	},
 
 	componentDidMount: function() {
 		// Ensures note gets saved if user abruptly quits the app
 		window.addEventListener( 'beforeunload', this.queueNoteSave.flush );
-	},
-
-	initializeNoteEditor: function( noteEditor ) {
-		this.noteEditor = noteEditor;
+		this.readNoteFromProps( this.props );
 	},
 
 	isValidNote: function( note ) {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function() {
+	hasContentChanged( editorState, note ) {
+		return get( note, 'data.content', '' )
+			!== editorState.getCurrentContent().getPlainText( '\n' )
+	},
+
+	handleEditorStateChange( editorState ) {
+		this.setState( { editorState } );
+		// only queue note save if it was the content that changed
+		if ( this.hasContentChanged( editorState, this.props.note ) ) {
+			this.queueNoteSave();
+		}
```

Comment:
There are a couple of notes here:

 - since the editor content is immutable, we should be able to detect if there were edits made by comparing `this.state.editorState.getCurrentContent()` to `editorState.getCurrentContent()`
 - let's make sure we are explicit about what sequencing we want here. the call to `this.queueNoteSave()` could actually occur before `setState()` has updated the component state. if we want this to happen in sequence, we can use the second argument to `setState()` and pass in a function to call once the state has updated